PMSCore/User.cs

```diff
@@ -1,9 +1,13 @@
+using System.Text.Json.Serialization;
+
 namespace PMSCore;
 
+[JsonConverter(typeof(UserJsonConverter))]
 public abstract class User : IUser
 {
-    protected User(string userId, HashSet<Permission> userPermissions)
+    protected User(UserType userType, string userId, HashSet<Permission> userPermissions)
     {
```

Comment:
protected User(string userId, HashSet<Permission> userPermissions)

and 

protected User(UserType userType, string userId, HashSet<Permission> userPermissions)

I think they are overlapping, maybe better to just use the second

Reply:
there is only protected User(UserType userType, string userId, HashSet userPermissions)